DllImportGenerator/readme.md

```diff
@@ -1,24 +1,33 @@
+Work on this project can be tracked and discussed via the [official issue](https://github.com/dotnet/runtime/issues/43060) in `dotnet/runtime`. The [P/Invoke source generator proposal](https://github.com/dotnet/runtime/blob/master/docs/design/features/source-generator-pinvokes.md) contains additional details.
 
-## Additional work
+## Example
 
-Below are additional work items that are not presently captured in this repository.
+The [Demo project](./DllImportGenerator/Demo) project is designed to be immediately consumable by everyone. It demonstrates a simple use case where the marshalling code is generated and a native function call with only [blittable types](https://docs.microsoft.com/dotnet/framework/interop/blittable-and-non-blittable-types) is made. A managed assembly with [native exports](./DllImportGenerator/TestAssets/NativeExports) is used in the P/Invoke scenario.
 
-### Required
+### Recommended scenarios:
 
+* Step into the `Demo` application and observe the generated code for the `Sum` functions.
+* Find the implementation of the `sumrefi` function and set a breakpoint. Run the debugger and explore the stack.
+* Add a new export in the `NativeExports` project and update the `Demo` application to call the new export.
+* Consider the differences and try the above scenarios when building in `Debug` or `Release`.
 
-* Add `GeneratedDllImportAttribute` to the BCL. See [`GeneratedDllImportAttribute.cs`](./Ancillary.Interop/GeneratedDllImportAttribute.cs).
-* Add support for calling [`SetLastError()`](https://docs.microsoft.com/windows/win32/api/errhandlingapi/nf-errhandlingapi-setlasterror) from managed code with the expected semantics. During P/Invoke transitions the runtime manipulates the associated thread's error code and thus breaks `SetLastError()` semantics.
+## Designs
 
-* APIs to handle [`SafeHandle`](https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.safehandle) manipulation.
+- [Code generation pipeline](./designs/Pipeline.md)
+- [Struct Marshalling](./designs/StructMarshalling.md)
 
-### Optional
+## Workflow
 
-* A tool to compare the resulting IL from the generated source to that generated by the built-in IL Marshaller system. This would help with validation of what is being generated.
+On Windows, loading the [solution](./DllImportGenerator.sln) in [Visual Studio](https://visualstudio.microsoft.com/) 2019 or later will enable the edit, build, debug inner dev loop. All features of Visual Studio are expected to operate correctly (e.g. Debugger, Test runner, Profiler).
 
-## Designs
+On non-Windows platforms, all features of the `dotnet` command line tool are supported for the respective project types (e.g. `build`, `run`, `test`).
 
-- [Code generation pipeline](./designs/Pipeline.md)
-- [Struct Marshalling](./designs/StructMarshalling.md)
+A consistent cross-platform inner dev loop is also supported in [Visual Studio Code](https://code.visualstudio.com/) when appropriate .NET extensions are loaded.
```

Comment:
Should we lead with the consistent cross-platform workflow (instead of starting with the Windows-specific one)?